qt/applications/workbench/workbench/plotting/plotscriptgenerator/__init__.py

```diff
@@ -206,12 +208,12 @@ def get_axes_object_variable(ax):
     # plt.subplots returns an Axes object if there's only one axes being
     # plotted otherwise it returns a list
     ax_object_var = AXES_VARIABLE
-    try:
-        if ax.numRows > 1:
-            ax_object_var += "[{row_num}]".format(row_num=row_num(ax))
-        if ax.numCols > 1:
-            ax_object_var += "[{col_num}]".format(col_num=col_num(ax))
-    except AttributeError:
-        # No numRows or NumCols members, so no list use the default
-        pass
+
+    if (LooseVersion('3.1.3') < LooseVersion(matplotlib.__version__) and ax.get_gridspec().nrows > 1) or \
```

Comment:
This feels a little messy. Is there a reason we can't just use the `hasattr` without needing to check the version?

```
if hasattr(ax, "get_gridspec"):
    num_rows = ax.get_gridspec().nrows
    num_cols = ax.get_gridspec().ncols
else:
    num_rows = ax.numRows
    num_cols = ax.numCols
...
```    
```

Reply:
Because it prints a deprecation warning, whilst it still has the attribute if that makes sense?

Reply:
If `get_gridspec` existed before the deprecation was added (I think it was? (https://github.com/matplotlib/matplotlib/commit/18d20e556ef94105869fe874ea52607c34c08b8d)), won't it always hit that first and not the `numRows` one there?

Reply:
Just tried it out and it seems fine from what I can tell. Let me know if the version fixes are ok.

Reply:
The CI has rediscovered the issue! 3.1.3 does have gridspec, but not nrows, and ncols.... I mean we can probably just do `hasattr(ax.get_gridspec(), nrows)` and `hasattr(ax.get_gridspec(), ncols)`.